doc/running.rst

```diff
@@ -46,6 +46,40 @@ environment variable, such as in the following example::
    $ srun -n 512 python -m mpi4py examples/wave-eager-mpi.py'
 
 
+OpenCL device selection
+-----------------------
+
+The :mod:`pyopencl` package supports selecting the device on which to run OpenCL code.
+There are multiple ways in which this selection can happen:
+
+- **Interactively.** When running mirgecom on the command line (i.e., not in a
+  batch script), it will present a (potentially multiple-choice) dialog with
+  the available OpenCL drivers (such as pocl) and the device (such as CPU or
+  GPU). After a device has been selected, pyopencl will print the chosen
+  device in the form of a value for the ``PYOPENCL_CTX`` environment
+  variable that can be used for subsequent executions.
+
+- **Through the PYOPENCL_CTX environment variable.** Pyopencl will use the
+  value of the ``PYOPENCL_CTX`` environment variable when available for device
+  selection. This value can either be list indices (such as ``0:1`` for first
+  driver, second device), or named abbreviations (such as ``port:tesla`` for pocl (=Portable OpenCL) and the first Nvidia Tesla GPU).
+
+- **Default device.** When not using one of the options above, pyopencl will
+  run on the first available device (which might not be the device you want)
+  by default.
```

Comment:
Discuss integration with https://github.com/majosm/squirm

cc @majosm

Reply:
I suspect squirm will mostly come into play when running the tests (+ maybe workflow stuff somewhere down the line?), so I think it can probably be omitted here. Once it gets integrated, I could maybe add a "Running the tests" section that describes how to set `SQUIRM_EXECUTOR_TYPE`, etc.

Reply:
Fair. Can't hurt to add a pointer, though. :)

Reply:
Something like #161 ?

Reply:
Yup!